lib/operations/estimated_document_count.js

```diff
@@ -2,32 +2,44 @@
 
 const Aspect = require('./operation').Aspect;
 const defineAspects = require('./operation').defineAspects;
-const CommandOperation = require('./command');
-const buildCountCommand = require('./common_functions').buildCountCommand;
-const handleCallback = require('../utils').handleCallback;
+const CommandOperationV2 = require('./command_v2');
 
-class EstimatedDocumentCountOperation extends CommandOperation {
+class EstimatedDocumentCountOperation extends CommandOperationV2 {
   constructor(collection, options) {
-    super(collection.s.db, options, collection);
+    super(collection, options);
+    this.collectionName = collection.s.namespace.collection;
   }
 
-  _buildCommand() {
-    const coll = this.collection;
-    let options = this.options;
+  execute(server, callback) {
+    const options = this.options;
+    const cmd = { count: this.collectionName };
```

Comment:
I thought we did not want to construct a command within `execute`. Should this be in a `buildCommand` method?

Reply:
The rule is not to build the command in the constructor. We explicitly want to build the command during execution, because this is the point in command construction where we have wire information for the server we're sending the command to (`buildCommand` is a refactoring that is called during execute). I haven't included a `buildCommand` yet because I'm not quite sure yet what it looks like, or if its necessary. I'd like to grow the design more organically here.